synth_waveform_F32.cpp

```diff
@@ -0,0 +1,92 @@
+#include <arm_math.h>
+#include "synth_waveform_F32.h"
+
+void AudioSynthWaveform_F32::update(void) {
+  audio_block_f32_t *block, *lfo;
+
+  if (_magnitude == 0.0f) return;
+
+  block = allocate_f32();
+  if (!block) return;
+
+  lfo = receiveReadOnly_f32(0);
+  switch (_OscillatorMode) {
+    case OSCILLATOR_MODE_SINE:
+        for (int i = 0; i < AUDIO_BLOCK_SAMPLES; i++) {
+          applyMod(i, lfo);
```

Comment:
You could consider moving your `applyMod` function outside of the `for` loop, since all of its inputs are known.  It's fine here, just something to consider.